tools/loader/ImageClassifier.cpp

```diff
@@ -189,6 +194,102 @@ void loadImagesAndPreprocess(const llvm::cl::list<std::string> &filenames,
       }
     }
   }
+
+  // For ONNX graphs with input in NHWC layout, we transpose the image data.
+  switch (imageLayout) {
+  case ImageLayout::NCHW:
+    break;
+  case ImageLayout::NHWC:
+    Tensor dataNHWC;
+    result->transpose(&dataNHWC, NCHW2NHWC);
+    *result = std::move(dataNHWC);
+    break;
+  }
+
+  // Convert the raw input to fp16.
+  if (convertInAndOutToFp16) {
+    result->convertToType(ElemKind::Float16Ty);
+  }
+}
+
+/// Write a prompt to stdout asking for filenames for classification. Read in
+/// those filenames and add them to \p filenames. \p filenames is cleared before
+/// adding the new set of filenames from stdin.
+static void getNextImageFilenames(std::vector<std::string> *filenames) {
+  // Clear out old filenames before adding new ones.
+  filenames->clear();
+
+  llvm::outs() << "Enter image filenames to classify: ";
+
+  // Add in each filename to the vector.
+  std::string filenamesRaw;
+  getline(std::cin, filenamesRaw);
+  std::istringstream iss(filenamesRaw);
+  std::string filename;
+  while (iss >> filename) {
+    filenames->push_back(filename);
+  }
+}
+
+/// Creates and \returns the protobuf loader given \p loader and the \p
+/// inputImageType. Note that this must come after loading images for inference
```

Comment:
nit: `\p` and `inputImageType` should be on the same line.

Reply:
Done